src/Okta.Sdk/Internal/UserAgentBuilder.cs

```diff
@@ -83,6 +83,31 @@ private string Generate()
             .Trim();
         }
 
+        private static string GetFrameworkDescription()
+        {
+            var frameworkDescription = RuntimeInformation.FrameworkDescription;
+
+            // RuntimeInformation.FrameworkDescription only returns the CLI version for .NET Core.
+            // We need this workaround to get the product version and have a useful report.
+            if (RuntimeInformation.FrameworkDescription.StartsWith(".NET Core", StringComparison.OrdinalIgnoreCase))
+            {
+                var runtimeAssembly = typeof(System.Runtime.GCSettings).GetTypeInfo().Assembly;
+
+                if (runtimeAssembly != null)
+                {
+                    var runtimeAssemblyLocation = runtimeAssembly.CodeBase.Split(new[] { '/', '\\' }, StringSplitOptions.RemoveEmptyEntries);
+                    int dotnetCoreAppIndex = Array.IndexOf(runtimeAssemblyLocation, "Microsoft.NETCore.App");
+
+                    if (dotnetCoreAppIndex > 0 && dotnetCoreAppIndex < runtimeAssemblyLocation.Length - 2)
```

Comment:
nit: Since `Array.IndexOf(...)` returns `-1` to indicate that the entry doesn't exist should we be considering `dotnetCoreAppIndex = 0` to be an allowed value in this bounds check? Or are we intentionally excluding the case where the path starts with `Microsoft.NETCore.App`?
```suggestion
                    if (dotnetCoreAppIndex >= 0 && dotnetCoreAppIndex < runtimeAssemblyLocation.Length - 2)
```

Reply:
Good catch!